integration_tests/src/main/python/join_test.py

```diff
@@ -182,6 +182,16 @@ def do_join(spark):
         return left.join(right, left.key == right.r_key, join_type)
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=_hash_join_conf)
 
+@validate_execs_in_gpu_plan('GpuShuffledHashJoinExec')
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', [all_basic_struct_gen], ids=idfn)
+@pytest.mark.parametrize('join_type', all_join_types, ids=idfn)
+def test_hash_join_struct(data_gen, join_type):
```

Comment:
Shouldn't we simply combine `test_hash_join_array`, `test_hash_join_map`, and `test_hash_join_struct` into a single `test_hash_join` that is parameterized with all the various types to test?  That matches the approach we use almost everywhere else in the tests and reduces the amount of boilerplate for this type of test.

Reply:
Thanks for the review. You are right, combining nested types test reduces the boilerplate code. I have combined other tests as well(SMJ, BHJ). Please take another look.